### PR TITLE
Fix some typos in plugin docs

### PR DESCRIPTION
**Pull request by andy-sweet**
I ran into these while working through the plugin docs. I have some other general comments that might lead to other changes, but I though it was worth making a tiny PR to fix these first.

### CODE REVIEW

File: plugins/stable/_npe2_writers_guide.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ by the corresponding writer contribution in the manifest).
    **attributes**, (e.g. `{'name': 'My Layer', 'opacity': 0.6}`)
 
    ```python
-   def single_layer_writer(path: str, data: Any, attributes: dict) -> Optional[str]:
+   def single_layer_writer(path: str, data: Any, attributes: dict) -> List[str]:
```
Comment (andy-sweet) [inline plugins/stable/_npe2_writers_guide.md:34]: I'm not sure about this one. The example below seems to indicate that a single layer writer that successfully writes should return a list. In my own plugin, I tried returning a list with one string, just the string, and `None`, and they all work. So maybe this should be `Union[List[str], str, None]`?

File: plugins/stable/_npe2_readers_guide.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ ReaderFunction = Callable[[PathOrPaths], List[LayerData]]
 
 def get_reader(path: PathOrPaths) -> Optional[ReaderFunction]:
     # If we recognize the format, we return the actual reader function
-    if isinstance(path, str) and path.endswith("*.xyz"):
+    if isinstance(path, str) and path.endswith(".xyz"):
```
Comment (andy-sweet) [inline plugins/stable/_npe2_readers_guide.md:37]: `str.endswith` just does plain string matching not glob-like matching, so using something like this in a plugin doesn't work (unless the filename actually contains an `*` before the `.xyz` I guess?).

**Comment by andy-sweet** [comment]
Sorry, wrong repo!

File: plugins/stable/_npe2_writers_guide.md (hunk 1 above)
Comment (nclack) [reply plugins/stable/_npe2_writers_guide.md:34]: This is definitely confusing. The part you highlighted disagrees with the type definition quoted just below this part.  In practice either works.

The intent was that line 34 is correct as is, so the type definition should get updated in the guide. This guide is generated from a [doc template in npe2](https://github.com/napari/npe2/blob/0cb4f0330d57425adb159345b3404844589820cd/_docs/templates/_npe2_writers_guide.md.jinja). So it needs to be corrected over there.

**Pull request closed without merging**